akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+};
+
+use super::*;
+use akd::directory::Directory;
+use akd::ecvrf::HardCodedAkdVRF;
+use akd::storage::memory::AsyncInMemoryDatabase;
+use akd::AkdLabel;
+use akd::AkdValue;
+use anyhow::Result;
+use aws_config::SdkConfig;
+use aws_sdk_s3::{
+    config::Builder as S3ConfigBuilder,
+    model::{Delete, ObjectIdentifier},
+    Client,
+};
+use aws_smithy_http::byte_stream::ByteStream;
+
+// These are constants that are matched in both this crate's `docker-compose.yml`
+// and the s3.yml workflow pipeline
+const ACCESS_KEY: &str = "test";
+const SECRET_KEY: &str = "someLongAccessKey";
+const TEST_REGION: &str = "us-east-2";
+const TEST_ENDPOINT: &str = "http://127.0.0.1:9000";
+
+macro_rules! bucket_name {
+    () => {{
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
+        name[..name.len() - 3]
+            .trim_end_matches("::{{closure}}")
+            .split("::")
+            .last()
+            .unwrap()
+            .replace("_", "")
+    }};
+}
+
+struct AuditInformation {
+    proof: akd::proof_structs::AppendOnlyProof<Hasher>,
+    phash: Digest,
+    chash: Digest,
+}
+
+/// Generate N audit proofs from a new tree
+async fn generate_audit_proofs(n: usize) -> Result<Vec<AuditInformation>, akd::errors::AkdError> {
+    let db = AsyncInMemoryDatabase::new();
+    let vrf = HardCodedAkdVRF {};
+    let akd = Directory::<_, _>::new::<Hasher>(&db, &vrf, false).await?;
+    let mut proofs = vec![];
+    // gather the hash + azks for epoch "0" (init)
+    let mut azks = akd.retrieve_current_azks().await?;
+    let mut phash = akd.get_root_hash::<Hasher>(&azks).await?;
+
+    for _epoch in 1..=n {
+        // generate (n) epochs of updates on the same user
+        let t_value = format!("certificate{}", _epoch);
+        akd.publish::<Hasher>(vec![(
+            AkdLabel::from_utf8_str("user"),
+            AkdValue::from_utf8_str(&t_value),
+        )])
+        .await?;
+        // generate the append-only proof
+        let proof = akd
+            .audit::<Hasher>((_epoch - 1) as u64, _epoch as u64)
+            .await?;
+        // update the azks + retrieve the "next" hash
+        azks = akd.retrieve_current_azks().await?;
+        let chash = akd.get_root_hash::<Hasher>(&azks).await?;
+
+        // we have everything we need, pass it along
+        let info = AuditInformation {
+            phash,
+            chash,
+            proof,
+        };
+        proofs.push(info);
+
+        // reset the "previous" hash for the next iteration
+        phash = chash;
+    }
+    Ok(proofs)
+}
+
+/// Flush the storage bucket of all blobs + delete the bucket (if it exists)
+async fn maybe_flush_storage(shared_config: &SdkConfig, bucket_name: &str) -> Result<()> {
+    // Build the S3 config from the shared SdkConfig
+    let config = S3ConfigBuilder::from(shared_config)
+        .retry_config(RetryConfig::disabled())
```

Comment:
I assume we're not using a retry since this is a test; is that right?

Reply:
yes, exactly. And with local network it shouldn't be a problem for testing scenarios